src/index.jsx

```diff
@@ -11,33 +11,39 @@ import { Redirect, Route, Switch } from 'react-router-dom';
 
 import { messages as headerMessages } from '@edx/frontend-component-header';
 
+import {
+  BaseComponent, UnAuthOnlyRoute, registerIcons, NotFoundPage, Logistration,
+} from './common-components';
 import configureStore from './data/configureStore';
-import { RegistrationPage } from './register';
-import { LoginPage } from './login';
 import {
   LOGIN_PAGE, PAGE_NOT_FOUND, REGISTER_PAGE, RESET_PAGE, PASSWORD_RESET_CONFIRM,
 } from './data/constants';
-import ForgotPasswordPage from './forgot-password';
-import {
-  BaseComponent, UnAuthOnlyRoute, registerIcons, NotFoundPage,
-} from './common-components';
-import ResetPasswordPage from './reset-password';
 import appMessages from './i18n';
-
 import './index.scss';
 
+import ForgotPasswordPage from './forgot-password';
+import { LoginPage } from './login';
+import { RegistrationPage } from './register';
+import ResetPasswordPage from './reset-password';
+
 registerIcons();
 
 subscribe(APP_READY, () => {
   ReactDOM.render(
     <AppProvider store={configureStore()}>
       <BaseComponent>
         <Switch>
-          <Route exact path="/">
-            <Redirect to={PAGE_NOT_FOUND} />
-          </Route>
-          <UnAuthOnlyRoute exact path={LOGIN_PAGE} component={LoginPage} />
-          <UnAuthOnlyRoute exact path={REGISTER_PAGE} component={RegistrationPage} />
+          <UnAuthOnlyRoute exact path="/" component={RegistrationPage} />
+          <UnAuthOnlyRoute
+            exact
+            path={LOGIN_PAGE}
+            render={() => <Logistration selectedPage={LOGIN_PAGE}><LoginPage /></Logistration>}
```

Comment:
Instead of passing the child component here can we do this in the `Logistration` component on the basis of `selectedPage`?